### PR TITLE
Add initial handling of BTM params

### PR DESCRIPTION
**Pull request by pinn3**

### CODE REVIEW

File: src/bite.v1.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 (function () {
     const state = {
-        programId: null
+        programId: null,
+        sessionId: null,
```
Comment (pinn3) [inline src/bite.v1.js]: This is not yet sent to the backend yet, it's just stored in the client side state

File: src/bite.v1.js [hunk 2]
```diff
@@ -21,9 +22,41 @@
...
+            }), {})
+
+        if (cookies['barnebys_session']) {
+            return cookies['barnebys_session']
+        }
+
+        const currentLocation = new URL(window.location)
+        const searchParams = new URLSearchParams(currentLocation.search)
+        const params = Array.from(searchParams.entries())
+        const btmTags = params
```
Comment (pinn3) [inline src/bite.v1.js]: This includes all the parameters starting with `btm_`, which means that we could handle `btm_locale` and other things as well, just not sure what to do with it just yet

File: src/bite.v1.js [hunk 3]
```diff
@@ -21,9 +22,41 @@
...
+    function determineSession() {
+        const cookies = document.cookie
+            .split('; ')
+            .map(cookie => cookie.split('='))
+            .reduce((mapping, [key, value]) => ({
+                ...mapping,
+                [key]: value,
+            }), {})
+
+        if (cookies['barnebys_session']) {
```
Comment (pinn3) [inline src/bite.v1.js]: We might want to add a TTL for any existing sessions as well, not sure what the desired behaviour is

File: src/bite.v1.js [hunk 4]
```diff
@@ -21,9 +22,41 @@
...
+            return cookies['barnebys_session']
+        }
+
+        const currentLocation = new URL(window.location)
+        const searchParams = new URLSearchParams(currentLocation.search)
```
Comment (pinn3) [inline src/bite.v1.js]: URL and URLSearchParams needs to be polyfilled in order to support older browsers

Comment (pinn3) [reply src/bite.v1.js]: It's not too necessary though, the values can be extracted simply using string methods/regex as well

Comment (pinn3) [reply src/bite.v1.js]: Replaced it with as custom function as it felt overkill to add a polyfill

File: src/bite.v1.js (hunk 3 above)
Comment (baboons) [reply src/bite.v1.js]: TTL should be a build env so we easily can configure that

Comment (pinn3) [reply src/bite.v1.js]: Sure thing, I'll add it

**Pull request merged**